### ISSUE TITLE
Feature flag to limit filters for online users

### ISSUE THREAD

**Issue by latin-panda** [medic/cht-core#8427]
**Describe the issue**
Make a performance analysis with big data to see the performance impact of applying multiple filters. Some questions we want to answer when having big data: 
- Is it more expensive to load reports with a default place filter? 
- What is the performance of applying more than one filter? 
    - They start with one and then add another one. How efficiently the app finds the intersection of the two sets of results and pages them?

It would also be interesting to compare the performance improvements when using CouchDB v3.x

**Describe the improvement you'd like**
If the performance is bad, we can limit the filter to apply just one at a time.

**Comment by garethbowen** [comment]
I don't think it's related to CouchDB because the filters are being applied in the browser, so CouchDB 3 won't save us here.

It should be reasonably easy to test if we use a full db and log in as an online user.

**Comment by latin-panda** [comment]
Yes, I understand it's on the front-end side. But since we're setting the DB for testing, it would be good to have numbers of the end-to-end and put them somewhere in the docs for future reference to start tracking these formally with version numbers. CouchDB 3 comment is out of curiosity only for comparison, if it's available.

**Comment by latin-panda** [comment]
<details>
<summary>The quick test I did the other day only covers the JavaScript side. It doesn't include network response times.</summary>

**With default filter**

- First load
  - CouchDB view by Place returns 80 000 reports 
  - Search from shared lib:
    - Get intersection: 93 ms
    - Do pagination: 24 ms
  - Total search time: 533 ms (from the view requesting search to receiving the results, not counting CouchDB processing) 

- After applying an additional search by term
  - CouchDB view by Place returns 80 000 reports 
  - CouchDB view by Search term returns 40 040 reports
  - Search from shared lib:
    - Get intersection: 92 ms
    - Do pagination: 0.40 ms
  - Total search time: 336 ms

- After applying a third filter by date: 
  - CouchDB view by Place returns 80 000 reports 
  - CouchDB view by Search term returns 40 040 reports
  - CouchDB view by Date returns 40 060 reports
  - Search from shared lib:
    - Get intersection: 201 ms
    - Do pagination: 0.39 ms
  - Total search time: 484 ms

**No default filter**

- First time Search + Date filter
  - CouchDB view by Search term returns 40 040 reports
  - CouchDB view by Date returns 40 060 reports
  - Search from shared lib:
    - Get intersection: 122 ms
    - Do pagination: 0.40 ms
  - Total search time: 538 ms

- Second time Search + Date filter
  - CouchDB view by Search term returns 40 040 reports
  - CouchDB view by Date returns 40 060 reports
  - Search from shared lib:
    - Get intersection: 131 ms
    - Do pagination: 0.5 ms
  - Total search time: 464 ms
  
</details>

**Comment by michaelkohn** [comment]
Is it useful to look at existing [telemetry](https://docs.communityhealthtoolkit.org/apps/guides/performance/telemetry/) (`search:reports:facilities`,`search:reports:facilities:search`, etc...) to gauge how long it takes in real projects?

<details>
<summary> SQL</summary>

```SQL
SELECT
	concat(doc#>>'{metadata,year}','-',doc#>>'{metadata,month}','-',COALESCE(doc#>>'{metadata,day}','1'))::date AS meta_date,
	doc#>>'{metadata,versions,app}' AS meta_version,
	doc#>>'{dbInfo,doc_count}' AS doc_count,

	'search:reports:facilities:search' AS metric,
	COALESCE((doc#>>'{metrics,search:reports:facilities:search,count}')::int,0) AS count,
	round((COALESCE((doc#>>'{metrics,search:reports:facilities:search,min}')::float,0)/1000)::numeric,2) AS min,
	round((COALESCE((doc#>>'{metrics,search:reports:facilities:search,max}')::float,0)/1000)::numeric,2) AS max
		    
FROM 
    couchdb_users_meta
    
WHERE
	doc->>'type'='telemetry'

ORDER BY 
	max DESC

```
</details>
